init.ts

```diff
@@ -50,6 +50,7 @@ async function onInitHooks() {
 	setSyncSettings({
 		'hw.device-type': {
 			resource: 'device_type',
+			// @ts-expect-error These typings need fixing :(
```

Comment:
But hwat is the problem?

Reply:
Basically these typings are hideously generic to the point it almost needs the same level of work as the pinejs-client did to make these also work. Since `resource` can be any `keyof Model` it means that `uniqueKey` needs to be a field of `Model[keyof Model]` and the only field that exists for all of them is `id` which means `slug` is rejected. So it needs to know the specific resource it's pointing at to look at the fields for that resource, and this has to work along with inference as well, it's not impossible to type but it's a real headache (with the `refersTo` version being even worse I think). I'll take another bash at these